lib/slack_logger/post_message.rb

```diff
@@ -8,19 +17,34 @@ def self.call(**args)
 
     def initialize(text:, channel: SlackLogger.config.channel, attachments: [])
       @text = text
-      @channel = channel
+      @channel = env_channel(channel)
       @attachments = attachments
     end
 
     def call
       return unless SlackLogger.config.enabled.call(channel: channel, text: text)
       client.chat_postMessage channel: channel, text: text, attachments: attachments
+    rescue Slack::Web::Api::Errors::ChannelNotFound
+      raise if SlackLogger.config.production?
+      logger.warn "attempt to send to non-existent slack channel #{channel.inspect}" if logger
```

Comment:
might be a good candidate for a nullobject so that you don't have to worry about if it exists or not.

Reply:
Would you suggest using a 3rd party gem like `Naught`? I am reluctant to bring in another dependency just for this

Reply:
I could alternatively lazily initialize it as `Logger.new(IO::NULL)`